deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/scala/org/deeplearning4j/spark/ml/impl/SparkDl4jNetworkWrapper.scala

```diff
@@ -0,0 +1,117 @@
+package org.deeplearning4j.spark.ml.impl
+
+import java.util
+
+import org.apache.spark.ml.{PredictionModel, Predictor}
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.ml.util._
+import org.apache.spark.mllib.linalg.Vector
+import org.apache.spark.mllib.regression.LabeledPoint
+import org.apache.spark.sql.Row
+import org.deeplearning4j.nn.conf.MultiLayerConfiguration
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork
+import org.deeplearning4j.optimize.api.IterationListener
+import org.deeplearning4j.spark.impl.multilayer.SparkDl4jMultiLayer
+import org.deeplearning4j.spark.ml.utils.{DatasetFacade, ParamSerializer}
+import org.deeplearning4j.util.ModelSerializer
+
+
+abstract class SparkDl4jNetworkWrapper[T, E <: SparkDl4jNetworkWrapper[T, E, M], M <: SparkDl4jModelWrapper[T, M]]
+(override val uid: String) extends Predictor[T, E, M]  {
+
+    protected var _multiLayerConfiguration : MultiLayerConfiguration = _
+    protected var _numLabels : Int = 2
+    protected var _freq : Int = 10
+    protected var _trainingMaster : ParamSerializer = _
+    protected var _listeners : util.Collection[IterationListener] = _
```

Comment:
I used util.Collection, since the dl4j library wants java's util library. I can use the implicit to make that happen though, if that seems cleaner.